aten/src/ATen/native/cpu/ReduceUtils.h

```diff
@@ -142,6 +152,19 @@ inline void update(scalar_t* out, scalar_t* data, int64_t K) {
       K);
 }
 
+template <typename scalar_t, ReductionType reduce,
+          typename std::enable_if_t<is_reduced_floating_point_v<scalar_t>, int> = 0>
+inline void update(at::opmath_type<scalar_t>* out, scalar_t* data, int64_t K) {
+  using opmath_t = at::opmath_type<scalar_t>;
+  using Vec = vec::Vectorized<opmath_t>;
+  map2<scalar_t, opmath_t>(
```

Comment:
i prefer to move the implementation of `map2` here and do not touch `functional_bfloat16.h`

Reply:
Done